src/detect.rs

```diff
@@ -0,0 +1,75 @@
+use crate::data::build_plan::BuildPlan;
+use crate::shared::{write_toml_file, BuildFromPath, Platform};
+use std::env;
+use std::path::PathBuf;
+use std::process;
+
+pub fn cnb_runtime_detect<P: Platform + BuildFromPath>(
+    detect_fn: impl Fn(DetectContext<P>) -> DetectResult,
```

Comment:
This never actually returns a `DetectResult` since this method will always `process::exit`. I keep going back and forth on this, but I wonder if we should return the enum instead of process exit.